app/assets/javascripts/plugins/analyticsPlugin.js

```diff
@@ -0,0 +1,167 @@
+    eventCategory: manifestId,
+    eventAction: type,
+    eventLabel: `${content}/${position}`,
+  }});
+}
+
+function* onUpdateCompanionWindow({ id, payload: { position, content }, type, windowId }) {
+  if (!windowId) return;
+
+  const { id: manifestId } = yield select(getManifest, { windowId });
+  const { content: existingContent } = yield select(getCompanionWindow, { companionWindowId: id });
+
+  yield put({ type: 'mirador/analytics', payload: {
+    hitType: 'event',
+    eventCategory: manifestId,
+    eventAction: type,
+    eventLabel: `${content || existingContent}/${position}`,
+  }});
+}
+
+
+function* onReceiveManifest({ manifestId, type }) {
+  yield put({ type: 'mirador/analytics', payload: {
+    hitType: 'event',
+    eventCategory: manifestId,
+    eventAction: type,
+  }});
+}
+
+function* onRequestSearch({ query, type, windowId }) {
+  const { id: manifestId } = yield select(getManifest, { windowId });
+
+  yield put({ type: 'mirador/analytics', payload: {
+    hitType: 'event',
+    eventCategory: manifestId,
+    eventAction: type,
+    eventLabel: query,
+  }});
+}
+
+function* onAddWindow({ type, window: { canvasId, manifestId } }) {
+  yield put({ type: 'mirador/analytics', payload: {
+    hitType: 'event',
+    eventCategory: manifestId,
+    eventAction: type,
+    eventLabel: canvasId,
+  }});
+}
+
+function* onMaximizeWindow({ type, windowId }) {
+  const { id: manifestId } = yield select(getManifest, { windowId });
+
+  yield put({ type: 'mirador/analytics', payload: {
+    hitType: 'event',
+    eventCategory: manifestId,
+    eventAction: type,
+  }});
+}
+
+function* onSetWindowViewType({ type, viewType, windowId }) {
+  const { id: manifestId } = yield select(getManifest, { windowId });
+
+  yield put({ type: 'mirador/analytics', payload: {
+    hitType: 'event',
+    eventCategory: manifestId,
+    eventAction: type,
+    eventLabel: viewType,
+  }});
+}
+
+function* onSelectAnnotation({ annotationId, type, windowId }) {
+  const { id: manifestId } = yield select(getManifest, { windowId });
+
+  yield put({ type: 'mirador/analytics', payload: {
+    hitType: 'event',
+    eventCategory: manifestId,
+    eventAction: type,
+    eventLabel: annotationId,
+  }});
+}
+
+function* onSetWorkspaceAction({ type }) {
+  yield put({ type: 'mirador/analytics', payload: {
+    hitType: 'event',
+    eventAction: type,
+  }});
+}
+
+function* onSetWorkspaceActionLayout({ layout, type }) {
+  yield put({ type: 'mirador/analytics', payload: {
+    hitType: 'event',
+    eventCategory: layout,
+    eventAction: type,
+  }});
+}
+
+function* onAddAuthRequest({ type, profile, windowId }) {
+  const { id: manifestId } = yield select(getManifest, { windowId });
+
+  yield put({ type: 'mirador/analytics', payload: {
+    hitType: 'event',
+    eventCategory: manifestId,
+    eventAction: type,
+    eventLabel: profile,
+  }});
+}
+
+function* sendAnalyticsEvent({ payload }) {
+  window.ga && window.ga('send', payload);
```

Comment:
Does this need to be:

```
window.ga('send',Object.values(payload));
```

Or does that automatically happen? Looking at https://developers.google.com/analytics/devguides/collection/analyticsjs/events

Reply:
Further down the doc, there's also:

```
ga('send', {
  hitType: 'event',
  eventCategory: 'Videos',
  eventAction: 'play',
  eventLabel: 'Fall Campaign'
});
```